spec/models/form526_submission_spec.rb

```diff
@@ -608,7 +608,10 @@
     context 'when the first name is NOT populated on the User' do
       before do
         # Ensure `subject` is called before stubbing `first_name` so that the auth headers are populated correctly
-        allow(User.find(subject.user_uuid)).to receive(:first_name).and_return nil
+        subject
+        user_with_nil_first_name = User.create(user)
+        allow(user_with_nil_first_name).to receive(:first_name).and_return nil
+        allow(User).to receive(:find).with(subject.user_uuid).and_return user_with_nil_first_name
```

Comment:
When a `User` for the `subject.user_uuid` is requested, provide our stubbed `user_with_nil_first_name` User record.
The previous code only stubbed on a specific instance resulting from the `User.find(subject.user_uuid)` call, which can be different than the instance returned from the `User.find(subject.user_uuid)` in the `get_first_name` method, even though the instances refer to the same `User` record.